bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/bridge/GroupePSABridgeHandler.java

```diff
@@ -0,0 +1,214 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.groupepsa.internal.rest.api.GroupePSAConnectApi;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.Vehicle;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.VehicleStatus;
+import org.openhab.binding.groupepsa.internal.rest.exceptions.GroupePSACommunicationException;
+import org.openhab.core.auth.client.oauth2.AccessTokenResponse;
+import org.openhab.core.auth.client.oauth2.OAuthClientService;
+import org.openhab.core.auth.client.oauth2.OAuthException;
+import org.openhab.core.auth.client.oauth2.OAuthFactory;
+import org.openhab.core.auth.client.oauth2.OAuthResponseException;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link GroupePSABridgeHandler} is responsible for handling commands,
+ * which are sent to one of the channels.
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+@NonNullByDefault
+public class GroupePSABridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(GroupePSABridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+    private static final long DEFAULT_POLLING_INTERVAL_M = TimeUnit.HOURS.toMinutes(1);
+
+    private final OAuthFactory oAuthFactory;
+
+    private @NonNullByDefault({}) OAuthClientService oAuthService;
+    private @Nullable ScheduledFuture<?> groupepsaBridgePollingJob;
+    private final HttpClient httpClient;
+
+    private @NonNullByDefault({}) String vendor;
+    private @NonNullByDefault({}) VendorConstants vendorConstants;
+    private @NonNullByDefault({}) String userName;
+    private @NonNullByDefault({}) String password;
+    private @NonNullByDefault({}) String clientId;
+    private @NonNullByDefault({}) String clientSecret;
+
+    private @NonNullByDefault({}) GroupePSAConnectApi groupePSAApi;
+
+    public GroupePSABridgeHandler(Bridge bridge, OAuthFactory oAuthFactory, HttpClient httpClient) {
+        super(bridge);
+        this.oAuthFactory = oAuthFactory;
+        this.httpClient = httpClient;
+    }
+
+    private void pollGroupePSAs() {
+        try {
+            List<Vehicle> vehicles = getVehicles();
+            if (vehicles != null) {
+                updateStatus(ThingStatus.ONLINE);
+                logger.debug("Found {} vehicles", vehicles.size());
+                for (Vehicle vehicle : vehicles) {
+                    logger.trace("Vehicle: {}", vehicle);
+                }
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging